notify_python35.cpp

```diff
@@ -0,0 +1,349 @@
+
+	// Set the enable flag
+        if (category.itemExists("enable"))
+        {
+                m_enabled = category.getValue("enable").compare("true") == 0 ||
+                            category.getValue("enable").compare("True") == 0;
+        }
+
+	// Check whether we have a Python 3.5 script file to import
+	if (category.itemExists(SCRIPT_CONFIG_ITEM_NAME))
+	{
+		try
+		{
+			// Get Python script file from "file" attibute of "scipt" item
+			m_pythonScript = category.getItemAttribute(SCRIPT_CONFIG_ITEM_NAME,
+								   ConfigCategory::FILE_ATTR);
+		        // Just take file name and remove path
+			std::size_t found = m_pythonScript.find_last_of("/");
+			if (found != std::string::npos)
+			{
+				m_pythonScript = m_pythonScript.substr(found + 1);
+			}
+		}
+		catch (ConfigItemAttributeNotFound* e)
+		{
+			delete e;
+		}
+		catch (exception* e)
+		{
+			delete e;
+		}
+	}
+
+	if (m_pythonScript.empty())
+	{
+		Logger::getLogger()->warn("Notification plugin '%s', "
+					  "called without a Python 3.5 script. "
+					  "Check 'script' item in '%s' configuration. "
+					  "Notification plugin has been disabled.",
+					  PLUGIN_NAME,
+					  this->getName().c_str());
+		// Force disable
+		this->disableDelivery();
+
+		return false;
+	}
+
+	return this->configure();
+}
+
+
+/**
+ * Call Python 3.5 notification method
+ *
+ * @param notificationName 	The name of this notification
+ * @param triggerReason		Why the notification is being sent
+ * @param message		The message to send
+ */
+bool NotifyPython35::notify(const std::string& deliveryName,
+			    const string& notificationName,
+			    const string& triggerReason,
+			    const string& customMessage)
+{
+	m_configMutex.lock();
+        if (!m_enabled)
+        {
+                // Release lock
+                m_configMutex.unlock();
+                // Current filter is not active: just return
+                return false;
+        }
+
+	// Save configuration variables and Python objects
+	string name = m_name;
+	string scriptName = m_pythonScript;
+	PyObject* method = m_pFunc;
+
+	// Release lock
+	m_configMutex.unlock();
+
+	Logger::getLogger()->error("_____ PYTHOn35 deliver ...");
+
+	// Call Python method passing an object
+	PyObject* pReturn = PyObject_CallFunction(method,
+						  "s",
+						  customMessage.c_str());
+
+	// Check return status
+	if (!pReturn)
+	{
+		// Errors while getting result object
+		Logger::getLogger()->error("Notification plugin '%s' (%s), script '%s', "
+					   "filter error, action: %s",
+					   PLUGIN_NAME,
+					   name.c_str(),
+					   scriptName.c_str(),
+					   "pass unfiltered data onwards");
+
+		// Errors while getting result object
+		logErrorMessage();
+	}
+	else
+	{
+		Logger::getLogger()->debug("PyObject_CallFunction() succeeded");
+
+		// Remove pReturn object
+		Py_CLEAR(pReturn);
+	}
+
+	return pReturn ? true : false;
+}
+
+/**
+ * Log current Python 3.5 error message
+ *
+ */
+void NotifyPython35::logErrorMessage()
```

Comment:
Please check logErrorMessage() implementation in python_plugin_interface.cpp. It has been fixed to print exception strings correctly.

Reply:
@AmandeepArora I'll create a new task for this